Replace valid with preview and viewer (#1452)
huggingface/datasets-server: docs/source/valid.mdx

Before you download a dataset from the Hub, it is helpful to know which datasets are available or if a specific dataset you're interested in is available. Datasets Server provides two endpoints for verifying whether a dataset is valid or not:

- `/valid` returns a list of all the datasets that work without any errors.
- `/is-valid` checks if a specific dataset works without any errors.

The API endpoints will return an error for datasets that cannot be loaded with the [🤗 Datasets](https://github.com/huggingface/datasets) library, for example, because the data hasn't been uploaded or the format is not supported.

<Tip warning={true}>
  The largest datasets are partially supported by Datasets Server. If they are{" "}

>   <a href="https://huggingface.co/docs/datasets/stream">streamable</a>, it can
>   extract the 100 first rows without downloading the whole dataset. This is
>   especially useful for previewing large datasets where downloading the whole
>   dataset may take hours! See the <code>preview</code> field in the response of{" "}
>   <code>/valid</code>.

Reviewer: ```suggestion
  <a href="https://huggingface.co/docs/datasets/stream">streamable</a>, Datasets Server can
  extract the first 100 rows without downloading the whole dataset. This is
  especially useful for previewing large datasets where downloading the whole
  dataset may take hours! See the <code>preview</code> field in the response of{" "}
  <code>/valid</code> to view these partially supported datasets.
```

</curl>
</inferencesnippet>

The endpoint response is a JSON containing lists of datasets nested under the following keys:

> - `viewer`: the dataset is fully-supported and the Dataset Viewer is working on the Hub dataset page. It also means that the dataset has been auto-converted to Parquet.

Reviewer: ```suggestion
- `viewer`: the dataset is fully-supported and the Dataset Viewer is working on the Hub dataset page. It also means the dataset has been auto-converted to Parquet.
```

> - `preview`: the dataset is partially supported. The Dataset Viewer on the Hub shows a Preview of the first 100 rows, obtained by streaming.

Reviewer: ```suggestion
- `preview`: the dataset is partially supported and the Dataset Viewer on the Hub dataset page shows a preview of the first 100 rows obtained by streaming.
```
